### PR TITLE
Feat: Added test utils and Storybook test runner

### PR DESCRIPTION
**Pull request by Giogia**
### Description

Added test render utils to provide default theme and remove Ant CSS dev only classes

Added Storybook test runner to check if components' stories run without errors

##### [INVOLVED_COMPONENT_HERE]

[CHANGES_HERE]

### Issue ticket number and link

[Link to the Jira Task](LINK_HERE)

### Checklist

- [ ] added the link to the Jira task
- [ ] commit message and branch name follow conventions
- [ ] tests are included
- [ ] changes are accessible and documented from components stories
- [ ] typings are updated or integrated accordingly with your changes
- [ ] all added files include Apache 2.0 license
- [ ] the changelog has been updated accordingly with your changes
- [ ] you are not committing extraneous files or sensible data
- [ ] the browser console does not have any logged errors

### CODE REVIEW

File: src/test-utils.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ */
+
+import { RenderOptions, RenderResult, render } from '@testing-library/react'
+import { ConfigProvider } from 'antd'
+import { ReactElement } from 'react'
+
+import { ThemeContext, defaultTheme } from './components/ThemeProvider'
+
+/**
+ * Wraps a React element with the a series of providers for testing purposes.
```
Comment (RiccardoCorona-MiaPlatform) [inline src/test-utils.tsx:26]:
```suggestion
 * Wraps a React element with a series of providers for testing purposes.
```

**Review by RiccardoCorona-MiaPlatform — DISMISSED** [review]

File: tsconfig.json [hunk 2]
```diff
@@ -20,7 +20,13 @@
     "strictNullChecks": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+
+    /* Tests */
+    "baseUrl": "src",
```
Comment (Danielecina) [inline tsconfig.json:26]: Come mai stiamo facendo degli alias? Così non rischiamo di confonderci? 
Sembra una lib esterna quando invece è una folder interna

Comment (Giogia) [reply tsconfig.json:26]: I don't think it creates confusion as the ctrl-click still brings the user to the local file.

It is also the React Testing library recommended setup: https://testing-library.com/docs/react-testing-library/setup/#configuring-jest-with-test-utils 

However, I'm open to other solutions, such as using @mia-platform-internal/testing-utils with a custom setup for this library.

File: .eslintrc.cjs [hunk 3]
```diff
@@ -14,7 +14,6 @@ module.exports = {
     "plugin:storybook/recommended"
   ],
   ignorePatterns: [
-    "dist",
```
[the remark this answers is not in the document]
Comment (Giogia) [reply .eslintrc.cjs:17]: Just moved to .eslintIgnore. I would keep this only for specific configuration files while using .eslintIgnore for directories.

[the remark this answers is not in the document]
Comment (mattia-fumo) [reply .eslintrc.cjs:17]: Clear

File: tsconfig.json (hunk 2 above)
Comment (Giogia) [reply tsconfig.json]: let's use the local path for now

File: jest.config.js [hunk 4]
```diff
@@ -15,6 +15,10 @@ export default {
     "jest-watch-typeahead/filename",
     "jest-watch-typeahead/testname"
   ],
+  moduleDirectories: [
+    'node_modules',
+    'src'
+  ],
```
[the remark this answers is not in the document]
Comment (Giogia) [reply jest.config.js]: removed :heavy_check_mark:

**Pull request merged**